PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-aad/src/main/java/com/microsoft/azure/toolkit/intellij/connector/aad/AzureUtils.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.intellij.connector.aad;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.microsoft.azure.toolkit.lib.Azure;
+import com.microsoft.azure.toolkit.lib.auth.AzureAccount;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+import com.microsoft.graph.authentication.TokenCredentialAuthProvider;
+import com.microsoft.graph.models.Application;
+import com.microsoft.graph.models.ApplicationAddPasswordParameterSet;
+import com.microsoft.graph.models.Domain;
+import com.microsoft.graph.models.PasswordCredential;
+import com.microsoft.graph.requests.GraphServiceClient;
+import okhttp3.Request;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+final class AzureUtils {
+    private AzureUtils() {
+    }
+
+    /**
+     * Creates a new client password credential for the given application.
+     *
+     * @param client      The Microsoft Graph client
+     * @param application The application to associate with the new credentials.
+     * @return The new credentials.
+     * @throws com.microsoft.graph.core.ClientException If the credentials could not be created
+     */
+    @NotNull
+    static PasswordCredential createApplicationClientSecret(@NotNull GraphServiceClient<Request> client,
+                                                            @NotNull Application application) {
+        assert application.id != null;
+
+        var credential = new PasswordCredential();
+        var secret = ApplicationAddPasswordParameterSet.newBuilder().withPasswordCredential(credential).build();
+        return client.applications(application.id)
+                .addPassword(secret)
+                .buildRequest()
+                .post();
```

Comment:
Although `Application` has a list of `PasswordCredentials`, these credentials don't contain the secret. https://docs.microsoft.com/en-us/graph/api/resources/passwordcredential?view=graph-rest-1.0#properties has a note about this for property `secretText`.
`secretText` is created on the server and the only way to fetch a `PasswordCredential` with a valid  `secretText`  is POST using `addPassword(...)`  here.

The code at https://github.com/microsoft/azure-tools-for-java/pull/5190/files#diff-80a5dbb90ea4214b15d318959d2345be0960acd5ca57aa0c08be4fdc476fcbd4R126 is calling the method from `AzureUtils` to fetch a `PasswordCredentials` instance with a `secretText` and then updated the `Application` model instance with it. A refresh request to the server wouldn't return the `secretText`.

The update of the `application` instance is mostly to keep all data for the form inside of the `application` and to avoid to sync another `secretText` property, which must be closely tied to the selected application.

@wangmingliang-ms Does this make sense? Should I still clarify or update the code?

Reply:
> Should I still clarify or update the code?
no, thanks.